### PR TITLE
Use named constants for event types

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: eventauth.go [hunk 1]
```diff
@@ -23,11 +23,19 @@ import (
...
+	invite                 = "invite"
+	public                 = "public"
+	mRoomCreate            = "m.room.create"
+	mRoomJoinRules         = "m.room.join_rules"
+	mRoomPowerLevels       = "m.room.power_levels"
+	mRoomMember            = "m.room.member"
+	mRoomThirdPartyInvite  = "m.room.third_party_invite"
+	mRoomAliases           = "m.room.aliases"
+	mRoomHistoryVisibility = "m.room.history_visibility"
+	mRoomRedaction         = "m.room.redaction"
```
Comment (Kegsay) [inline eventauth.go]: Can these be public please or else I'll be repeating this in Dendrite code.

**Comment by Kegsay** [comment]
Otherwise LGTM.

File: eventauth.go (hunk 1 above)
Comment (NegativeMjark) [inline eventauth.go]: Done 258b2b4

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**